canonical/ubuntu.com
# Rebrand certified configuration search result pages

## Done

This PR rebrands the search pages under ubuntu.com/certified. 

Wireframes: https://www.figma.com/design/1gDRefkvA5TnZGWI9tdkZB/24.10-Certified-hardware?node-id=55-202&p=f&t=hmRyTRDr0D2ki2zG-0

#### Change structure

To help with the review, please check out the description of the changes below.

Firstly, we need to define what are all the pages that allow users to search for hardware. Currently we have three different search page types:
* generic search results page, leads if you go to ubuntu.com/certified?q=Dell
* category search pages:
    - `/certified/laptops` (ubuntu.com/certified/laptops)
    - `/certified/desktops` (ubuntu.com/certified/desktops)
    - `/certified/iot` (ubuntu.com/certified/iot)
    - `/certified/servers` (ubuntu.com/certified/servers)
    - `/certified/socs` (ubuntu.com/certified/socs)
* vendor search pages: under: `/certified/vendor/HP` (ubuntu.com/certified/HP) (and a few other vendors)

Previously, the generic search results page was fully separately defined in the `search-results.html` template, while the other functionalities shared the same search results table. I have now moved all functionalities to share the same search results table to avoid code duplication.

All search result templates have the same file structure (such files are `search-results.html`, `laptops.html`, `servers.html`, `desktops.html`, `devices.html`, `socs.html` and `vendors/vendor.html`):
1. Hero section (different for each search page)
2. Horizontal line
3. Hardware results shown in a table

The hardware results block contains a few small files to break them down. I have made this change to make them easier to review. Below is a hierarchy of the file structure:
![Screenshot from 2024-11-04 15-14-07](https://github.com/user-attachments/assets/e59cbc01-3e8b-4127-800e-40943db893ca)

## QA

- Check out this feature branch
- Run the site using the command `./run serve` or `dotrun`
- View the site locally in your web browser at: http://0.0.0.0:8001/
    - Be sure to test on mobile, tablet and desktop screen sizes
- Demo link: https://ubuntu-com-14381.demos.haus/

## Issue / Card

Fixes #

## Screenshots

#### Generic search results

![Screenshot from 2024-11-04 15-23-43](https://github.com/user-attachments/assets/fe12bdfa-89e4-4ef3-8f01-747f6c0ed4ad)

#### Laptops

![Screenshot from 2024-11-04 15-23-58](https://github.com/user-attachments/assets/175e0c73-f870-4c7c-a140-ac4568c1d337)

#### Desktops

![Screenshot from 2024-11-04 15-24-18](https://github.com/user-attachments/assets/ab0fa78d-e35d-4548-906c-ba055ec6a015)

#### IoT

![Screenshot from 2024-11-04 15-25-10](https://github.com/user-attachments/assets/0732175e-c3ab-47ae-991f-98bff46b6a43)


#### Servers

![Screenshot from 2024-11-04 15-25-36](https://github.com/user-attachments/assets/ebff59f5-77bc-4d1b-9909-ab4d744d8469)

#### System on Chip (SoC)

![Screenshot from 2024-11-04 15-25-23](https://github.com/user-attachments/assets/a064910e-ffb8-4582-89b0-29a1a3d26660)

#### Vendor

![image](https://github.com/user-attachments/assets/19c966f2-9cbd-438e-a626-7a30d3bff8fc)

## Help

[QA steps](https://discourse.canonical.com/t/qa-steps/152) - [Commit guidelines](https://discourse.canonical.com/t/commit-guidelines/148)
